frontend_tests/node_tests/stream_list.js

```diff
@@ -96,9 +97,9 @@ set_global('topic_list', {});
     stream_list.build_stream_list();
 
     var expected_elems = [
-        split,
-        devel_sidebar,
-        social_sidebar,
+        devel_sidebar,          //pinned
+        split,                  //separator
+        social_sidebar,         //not pinned
     ];
```

Comment:
Can you add a test section for the corner case you just fixed?

Reply:
Sure, I added tests for two corner cases (only pinned and dormant streams and solely pinned streams).